File: C++/002. Add Two Numbers.cpp
class Solution {
public:
    ListNode* addTwoNumbers(ListNode* l1, ListNode* l2) {
        ListNode* result;
        result = new ListNode(-1);
        int val=0;
        if(l1==NULL && l2==NULL)
        {
            return NULL;
        }
        else if(l1==NULL)
        {
            return l2;
        }
        else if(l2==NULL)
        {
            return l1;
        }
        else
        {
            result->next=l1;
            val = l1->val + l2->val;
            l1->val = val%10;
            if(val>=10)
            {
                if(l1->next==NULL)
                {
                    l1->next = new ListNode(1);
                }
                else if(l2->next==NULL)
                {
                    l2->next = new ListNode(1);
                }
                else
                {
                    l1->next->val = l1->next->val+1;
                }
            }
            result->next->next=addTwoNumbers(l1->next, l2->next);
        }
        return result->next;
    }
};

File: python/002. Add Two Numbers.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.next = None
class Solution(object):
    def addTwoNumbers(self, l1, l2):
        """
        :type l1: ListNode
        :type l2: ListNode
        :rtype: ListNode
        """
        carry = 0
        head = ListNode(0)
        cur = head 
        while l1 or l2:
            cnt = 0
            if l1: 
                cnt += l1.val
                l1 = l1.next
            if l2: 
                cnt += l2.val
                l2 = l2.next
            cnt += carry
            carry = cnt//10
            cnt %= 10
            cur.next = ListNode(cnt)
            cur = cur.next
            
        if carry: cur.next = ListNode(carry)
        return head.next

